tests/network.py

```diff
@@ -123,14 +141,48 @@ def destroy(self):
     def run(self, command, user=os.getenv("USER")):
         """
         Executes a command under the given user from this virtual node. Returns
-        an NSOpen object to control the process. NSOpen has the same API as
-        subprocess.POpen.
+        a context manager that returns NSOpen object to control the process.
+        NSOpen has the same API as subprocess.POpen.
+        """
+        sudo_command = ['sudo', '-E', '-u', user,
+                        'env', 'PATH=' + os.getenv("PATH")] + command
+        return managed_nspopen(self.namespace, sudo_command,
+                               stdin=subprocess.PIPE, stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE, universal_newlines=True,
+                               start_new_session=True)
+
+    def wait_or_timeout_command(self, command, name, timeout):
```

Comment:
What about naming this just `wait` or something in that spirit? The timeout raises an exception so I don't think it's very good to have it in the name... (in the name means it's expected, means it's not an exception?)